test/GovernorCountingFractional.t.sol

```diff
@@ -249,35 +264,37 @@ contract GovernorCountingFractionalTest is Test {
     return voteSplit.percentFor > 0 || voteSplit.percentAgainst > 0 || voteSplit.percentAbstain > 0;
   }
 
-  function _castVotes(Voter[4] memory voters, uint256 _proposalId) internal {
-    Voter memory voter;
-    for (uint8 _i = 0; _i < voters.length; _i++) {
-      voter = voters[_i];
-
-      assert(!governor.hasVoted(_proposalId, voter.addr));
+  function _castVotes(Voter memory _voter, uint256 _proposalId) internal {
+    if (_voter.weight == 0) return;
+    assertEq(governor.hasVoted(_proposalId, _voter.addr), false);
 
-      bytes memory fractionalizedVotes;
-      FractionalVoteSplit memory voteSplit = voter.voteSplit;
+    bytes memory fractionalizedVotes;
+    FractionalVoteSplit memory voteSplit = _voter.voteSplit;
 
-      if (_isVotingFractionally(voteSplit)) {
-        fractionalizedVotes = abi.encodePacked(
-          uint128(voter.weight.mulWadDown(voteSplit.percentFor)),
-          uint128(voter.weight.mulWadDown(voteSplit.percentAgainst)),
-          uint128(voter.weight.mulWadDown(voteSplit.percentAbstain))
+    if (_isVotingFractionally(voteSplit)) {
+      fractionalizedVotes = abi.encodePacked(
+        uint128(_voter.weight.mulWadDown(voteSplit.percentFor)),
+        uint128(_voter.weight.mulWadDown(voteSplit.percentAgainst)),
+        uint128(_voter.weight.mulWadDown(voteSplit.percentAbstain))
+      );
+      vm.expectEmit(true, true, true, true);
+      emit VoteCastWithParams(
+        _voter.addr, _proposalId, _voter.support, _voter.weight, "Yay", fractionalizedVotes
         );
-        vm.expectEmit(true, false, false, true);
-        emit VoteCastWithParams(
-          voter.addr, _proposalId, voter.support, voter.weight, "Yay", fractionalizedVotes
-          );
-      } else {
-        vm.expectEmit(true, false, false, true);
-        emit VoteCast(voter.addr, _proposalId, voter.support, voter.weight, "Yay");
-      }
+    } else {
+      vm.expectEmit(true, true, true, true);
+      emit VoteCast(_voter.addr, _proposalId, _voter.support, _voter.weight, "Yay");
+    }
+
+    vm.prank(_voter.addr);
+    governor.castVoteWithReasonAndParams(_proposalId, _voter.support, "Yay", fractionalizedVotes);
 
-      vm.prank(voter.addr);
-      governor.castVoteWithReasonAndParams(_proposalId, voter.support, "Yay", fractionalizedVotes);
+    assertEq(governor.hasVoted(_proposalId, _voter.addr), true);
```

Comment:
Nit, you can use `assertTrue(governor.hasVoted(_proposalId, _voter.addr))` instead. Feel free to resolve this comment yourself.

Reply:
Thanks. Done in b202e74